File: Go/백준/Gold/12784. 인하니카 공화국/인하니카 공화국.go
package main

import (
	"bufio"
	"os"
	"strconv"
)

type Node struct {
	idx, cost int
}

func (n Node) unpack() (int, int) {
	return n.idx, n.cost
}

func dfs(tree [][]Node, prev int, now int) int {
	res := 0
	for _, node := range tree[now] {
		nxt, cost := node.unpack()
		if nxt != prev {
			res += min(cost, dfs(tree, now, nxt))
		}
	}
	if res == 0 {
		return 1e9
	}
	return res
}

func solve() {
	N, M := intInput(), intInput()
	if N == 1 {
		bw.WriteString("0")
		return
	}
	tree := make([][]Node, N+1)
	for i := 1; i < N+1; i++ {
		tree[i] = make([]Node, 0)
	}
	for i := 0; i < M; i++ {
		a, b, c := intInput(), intInput(), intInput()
		tree[a] = append(tree[a], Node{b, c})
		tree[b] = append(tree[b], Node{a, c})
	}

	bw.WriteString(strconv.Itoa(dfs(tree, 1, 1)))
}

func main() {
	bs.Split(bufio.ScanWords)
	defer bw.Flush()

	tc := intInput()
	for t := 0; t < tc; t++ {
		solve()
		bw.WriteString("\n")
	}
}

var (
	bs = bufio.NewScanner(os.Stdin)
	bw = bufio.NewWriter(os.Stdout)
)

func input() string {
	bs.Scan()
	return bs.Text()
}

func intInput() int {
	num, _ := strconv.Atoi(input())
	return num
}

File: Python/백준/Gold/12784. 인하니카 공화국/인하니카 공화국.py
import sys; input = sys.stdin.readline


def solve():
    def dfs(prev=1, now=1):
        res = 0
        for nxt, cost in tree[now]:
            if nxt == prev: continue
            res += min(dfs(now, nxt), cost)
        return res if res else float('inf')


    N,M = map(int,input().split())
    if N == 1: return 0
    tree = [[] for _ in range(N+1)]
    for _ in range(M):
        a,b,c = map(int,input().split())
        tree[a].append((b,c))
        tree[b].append((a,c))
    return dfs()

for _ in range(int(input())):
    print(solve())